docker-compose-3-nodes.yaml

```diff
@@ -0,0 +1,387 @@
+# Copyright 2018 Intel Corporation
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+# ------------------------------------------------------------------------------
+
+version: '3.6'
+
+volumes:
+  cargo-registry:
+  cargo-registry-tp:
+  poet-share:
+
+services:
+
+  validator-1:
+    image: sawtooth-poet-validator:latest
+    build:
+      context: .
+      dockerfile: poet-validator.dockerfile
+      args:
+        - http_proxy
+        - https_proxy
+        - no_proxy
+    expose:
+      - 4004
+      - 8800
+      - 5050
+    volumes:
+      - ./logs1/validator:/var/log/sawtooth
+      - poet-share:/poet_keys
+    # start the validator with an empty genesis batch
+    working_dir: /root
+    command: "bash -c \"\
+        rm -f /poet_keys/poet-config.batch && \
+        sawadm keygen && \
+        sawset genesis \
+          -k /etc/sawtooth/keys/validator.priv \
+          -o config-genesis.batch && \
+        sawset proposal create \
+          -k /etc/sawtooth/keys/validator.priv \
+          sawtooth.consensus.algorithm=poet2 \
+          sawtooth.poet.block_claim_delay=0 \
+          -o config.batch && \
+        cp /etc/sawtooth/keys/validator.priv /poet_keys/validator-1.priv && \
+        cp /etc/sawtooth/keys/validator.pub /poet_keys/validator-1.pub && \
```

Comment:
Let's keep this now as a temporary solution. Copying validator public key to poet engine won't work if poet engine and validator are running on different machines. I saw API get_peers_public_keys() in consensus SDK. See if we can use it to get public key of validator.

Reply:
Ok